sdk/nodejs/output.ts

```diff
@@ -460,6 +463,58 @@ function getResourcesAndDetails<T>(allOutputs: Output<Unwrap<T>>[]): [Resource[]
     return [allResources, isKnown, isSecret];
 }
 
+/**
+ * Unknown represents a value that is unknown.
+ */
+export class Unknown {
+     /**
+     * @internal
+     * A private field to help with RTTI that works in SxS scenarios.
+     *
+     * This is internal instead of being truly private, to support mixins and our serialization model.
+     */
+    // tslint:disable-next-line:variable-name
+    public readonly __pulumiUnknown: boolean = true;
+
+    /**
+     * Returns true if the given object is an instance of Unknown. This is designed to work even when
+     * multiple copies of the Pulumi SDK have been loaded into the same process.
+     */
+    public static isInstance(obj: any): obj is Unknown {
+        return utils.isInstance(obj, "__pulumiUnknown");
+    }
+}
+
+/**
+ * unknown is the singleton unknown value.
+ */
+export const unknown = new Unknown();
+
+/**
+ * isUnknown returns true if the given value is unknown.
+ */
+export function isUnknown(val: any): boolean {
+    return Unknown.isInstance(val);
+}
+
+/**
+ * containsUnknowns returns true if the given value is or contains unknown values.
+ */
+export function containsUnknowns(val: any): boolean {
```

Comment:
consider not exporting this.

Reply:
From an SDK perspecitve, this looks like a fine shape.  From an impl, i would have to go over this with a fine-toothed comb.